### DISCUSSION TITLE
Getting started with contributions

### DISCUSSION THREAD

**Discussion by wlbaker** [serenity-bdd/serenity-core#3362] [General]
This is a large and dynamic code base.  I'd like to make local changes in the code base and recompile, but it looks possibly difficult to do that.  I've learned that the mvn build process is more reliable than the gradle build.  However, I don't know how to handle versions and snapshots in MVN.  I would normally expect to publish to a local repository (./gradlew publishToMavenLocal) and set the version number (4.0.29-SNAPSHOT-wlb) in only one place.  However, this does not look easy to do in the current config.  Each subproject has it's own version tag (all set to 4.0.29) and the snapshot repo is set to sonotype.

Is there an easy way to create a local snapshot build which all the subprojects will use?  Changing all the pom.xm files to use my local version...that feels like the wrong thing to do.

If I'm struggling with this, probably other people are as well.  Best way to get new contributors is to make the build and contribution process easy.  A one-page HTML doc on this subject might pay dividends.

**Comment by wakaleo** [comment]
Thanks for your feedback! The Gradle build was abandoned a long time ago, when the Artifactory repository was shut down, so currently the Maven build is effectively the only build process to use. To build a local snapshot you run `mvn install -DskipTests` (assuming you don't want to run all the tests locally as this takes some time). This will build a snapshot version and publish it to the maven local repo.

I will try to write up some doc soon, it is a good idea.
